src/Symfony/Component/Security/Http/EventListener/RememberMeListener.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Symfony\Component\Security\Http\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\RememberMeBadge;
+use Symfony\Component\Security\Http\Event\LoginFailedEvent;
+use Symfony\Component\Security\Http\Event\LoginSuccessEvent;
+use Symfony\Component\Security\Http\RememberMe\RememberMeServicesInterface;
+
+/**
+ * The RememberMe *listener* creates and deletes remember me cookies.
+ *
+ * Upon login success or failure and support for remember me
+ * in the firewall and authenticator, this listener will create
+ * a remember me cookie.
+ * Upon login failure, all remember me cookies are removed.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @final
+ * @experimental in 5.1
+ */
+class RememberMeListener implements EventSubscriberInterface
+{
+    private $rememberMeServices;
+    private $logger;
+
+    public function __construct(RememberMeServicesInterface $rememberMeServices, ?LoggerInterface $logger = null)
+    {
+        $this->rememberMeServices = $rememberMeServices;
+        $this->logger = $logger;
+    }
+
+    public function onSuccessfulLogin(LoginSuccessEvent $event): void
+    {
+        $passport = $event->getPassport();
+        if (null === $passport) {
```

Comment:
I just realized that `$event->getPassport()` return `?PassportInterface`... which goes all the way back to `AuthenticatorManager::handleAuthenticationSuccess()`, which allows a `?PassportInterface` argument. I don't think any of this is needed - if you look at the `AuthenticatorManager` logic, we DO have a `PassportInterface` in all cases.

Reply:
This is a leftover - I created `AnonymousPassport` when I realized it was too annoying to have `?PassportInterface` everywhere

I'll update